### ISSUE TITLE
Homebridge v2.0 support

### ISSUE THREAD

**Issue by hanwalt** [SeydX/homebridge-tado-platform#150]
Although it is unclear when Homebridge v2.0 will be released I am concerned that some plugins are not ready vor v2 and will break up. As I read in another issue homebridge-tado-platform will break with v2.0

I know that @seydx didn`t update the plugin quite a long time (maybe he sold his Tado stuff?) but there is still hope that there will be an updated version because the plugin integration in Homekit is much better than the native one!

So please dear @seydx be so kind and do some favor for some eager homebridge-tado-platform users!

**Comment by badgertastic** [comment]
Would also like HB v2 compatibility please. Thanks

**Comment by UBEXHH** [comment]
Last update of the plugin was May 2021 and there are 57 open issues, I think Seydx is maybe dead?!

**Comment by MarcZ2202** [comment]
I would even pay for it

**Comment by mariosc** [comment]
Yes, an updated version is highly appreciated

**Comment by adavison007** [comment]
Has anyone tried it on Homebridge 2.0... some plugins will work without the support depending on the code used... I'm too scared to update and try, as most of my plugins are not yet supported... although they maybe fine as is.

**Comment by PaulPizzey247** [comment]
@adavison007 Yes I've tried it and the plugin doesn't work, just throws the below error and keeps restarting t he child bridge. Frustratingly, all my other plugins work fine, even the ones that aren't confirmed to work except this one.

**_[tadoPlatform] Child bridge ended (code 1, signal null). The child bridge ended unexpectedly, which is normally due to the plugin not catching its errors properly. Please report this to the plugin developer by clicking on the 'Report An Issue' option in the plugin menu dropdown from the Homebridge UI. If there are related logs shown above, please include them in your report._**

**Comment by adavison007** [comment]
> [@adavison007](https://github.com/adavison007) Yes I've tried it and the plugin doesn't work, just throws the below error and keeps restarting t he child bridge. Frustratingly, all my other plugins work fine, even the ones that aren't confirmed to work except this one.
> 
> **_[tadoPlatform] Child bridge ended (code 1, signal null). The child bridge ended unexpectedly, which is normally due to the plugin not catching its errors properly. Please report this to the plugin developer by clicking on the 'Report An Issue' option in the plugin menu dropdown from the Homebridge UI. If there are related logs shown above, please include them in your report._**

Thanks for the update... Nightmare... @seydx PLEASE can you fix?

**Comment by PaulPizzey247** [comment]
@adavison007 I really hope @seydx does update the plugin, but as it hasn't been updated since 2021, I don't hold much hope.

Hopefully someone else is developing their own plugin.

**Comment by smart7324** [comment]
Hey, I just downloaded Homebridge v2.0.0-alpha.37 and also got the same error. I did some debugging and was able to fix the crash.

Homebridge 2.0 is using HAP-NodeJS v1 and it seems that they changed the name of the Battery Service from `BatteryService` to `Battery`. This is the only reason why the plugin is crashing on startup with `Child bridge ended (code 1, signal null)`.

To fix it, we only need to replace all occurrences of `this.api.hap.Service.BatteryService` in the plugin code with `this.api.hap.Service.Battery`.

To find all occurrences, we can use a grep command with the path of your plugin location of your homebridge instance: 

```bash
grep -irn '/var/lib/homebridge/node_modules/homebridge-tado-platform/' -e 'this.api.hap.Service.BatteryService'
```

These are the files we need to change:

```
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/thermostat.js:47:    let batteryService = this.accessory.getService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/thermostat.js:52:        batteryService = this.accessory.addService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/temperature.js:38:    let batteryService = this.accessory.getService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/temperature.js:43:        batteryService = this.accessory.addService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/humidity.js:38:    let batteryService = this.accessory.getService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/humidity.js:43:        batteryService = this.accessory.addService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/contact.js:44:    let batteryService = this.accessory.getService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/contact.js:49:        batteryService = this.accessory.addService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/heatercooler.js:58:    let batteryService = this.accessory.getService(this.api.hap.Service.BatteryService);
/var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/heatercooler.js:63:        batteryService = this.accessory.addService(this.api.hap.Service.BatteryService);
```

Save all changes and (re)start your child bridge. The plugin will then just work fine with Homebridge 2.0:

![Image](https://github.com/user-attachments/assets/4708b8d8-84d0-424a-a15e-88e7fdd3e477)

**Another important note:** On 21st of March 2025 the plugin will stop working in its current state as tado is no longer allowing the authentication to their API using the password grant flow. I found a workaround that I implemented to make it working again. I wrote a detailed guide in this issue: https://github.com/seydx/homebridge-tado-platform/issues/152

**Comment by adavison007** [comment]
> var/lib/homebridge/node_modules/homebridge-tado-platform/src/accessories/

Many thanks, this is great, HELP PLEASE! I can't locate my Tado directory? I do not have this location? /var/lib/homebridge/node_modules/homebridge-tado-platform/ 

Infact, I don't have /var/lib/homebridge/node_modules/

In /var/lib/homebridge/ I show the following. (silver hollow is my house name in Tado) 

<img width="2209" alt="Image" src="https://github.com/user-attachments/assets/186f9d90-9038-46ba-97e7-b2c2f600c873" />

**Comment by smart7324** [comment]
I will create a fork of the plugin. See #152

**Comment by adavison007** [comment]
> I will create a fork of the plugin. See [#152](https://github.com/seydx/homebridge-tado-platform/issues/152)

BRILLIANT, YOU ARE A STAR!

**Comment by smart7324** [comment]
Fork is released. Find all details and instructions here: https://github.com/smart7324/homebridge-tado-platform-next/issues/1

**Comment by smart7324** [comment]
Fork has been merged to this repository. This plugin supports Homebridge v2.0 => closed

**Comment by MarcZ2202** [comment]
Wow, unbelievable. Thank you so much!!!
